builder: install option now copies files contents instead of symlinking

sevki opened the pull request:
you can also set the build_out folder by setting the BUILD_OUT
env var from either command line or by editing the .build file

Signed-off-by: Sevki s@sevki.org

sevki committed:
targets/harvey: added old_build option

Signed-off-by: Sevki <s@sevki.org>

sevki committed:
builder: install now copies files contents instead of symlinking

you can also set the build_out folder by setting the BUILD_OUT
env var from either command line or by editing the .build file

Signed-off-by: Sevki <s@sevki.org>

sevki committed:
Merge branch 'master' into install

Signed-off-by: Sevki Hasirci <s@sevki.org>

hdonnay commented on builder/coordinator.go:
```diff
@@ -265,3 +227,73 @@ func (b *Builder) visit(n *Node) {
+			util.GetProjectPath(),
+			"build_out",
+		)
+	}
```
If this is variable, it probably shouldn't be hard-coded in the `util.Hash` functions.

sevki committed:
util: make an exception for the build_out and env set variables

there is now a function under util.BuildOut() that returns the
correct build output folder.

I've also move the GetEnv function to util as it made more sense
there. There is stilla build.Getenv function that wraps the util
behaviour that should be fixed with gofmt

Signed-off-by: Sevki <s@sevki.org>

sevki committed:
target: deprecate build.Getenv in favour of util.Getenv

gofmt rocks!

Signed-off-by: Sevki <s@sevki.org>

sevki committed:
Merge branch 'install' of github.com:sevki/build into install

Signed-off-by: Sevki <s@sevki.org>

hdonnay commented:
This now depends on go1.7 because of `context`. I'm cool with this, but travis is not.

sevki committed:
ci: only support 1.7> for context stdlib support

Signed-off-by: Sevki <s@sevki.org>

hdonnay commented:
LGTM

hdonnay commented:
Okay, once more with feeling:

LGTM

[![Approved with PullApprove](https://img.shields.io/badge/Harvey--OS/core-approved-brightgreen.svg)](https://pullapprove.com/sevki/build/pull-request/31/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=sevki/build)
